### PR TITLE
odpi/egeria-ui#480 Support for new ui

### PR DESCRIPTION
**Pull request by planetf1**
Signed-off-by: Nigel Jones <nigel.l.jones+git@gmail.com>

Adds support for new Egeria UI
 - remove additional 'nginx' container previously used for distributing traffic between static ui & api backend
 - add ssl config to staticui pod (uses old config, which itself should be updated - but that's a bigger issue...)
 - add config for ui api server
 - fix definitions for using ssl on port 8443
 
 Todo
  - get calls to backend working
  - cleanup some more unused definitions (ie env)
  
See odpi/egeria-ui#480

Notes
 * This will only work when the new ui image is used
 * version parm will be updated once testing complete - to a 3.14 prerelease
 * this depends on odpi/egeria-ui#481 being merged, AND a new Docker image being pushed (ie a version increment)
 * as above - old, hardcoded, self-signed ssl certs. needed to get a step forward, but as discussed in our planning sessions, something we need to complete the work on.

### CODE REVIEW

**Comment by planetf1** [comment]
@lpalashevski @sarbull

This chart change, with it's prereq docker change, is an initial attempt to get the new UI integrated.

However I am unclear on the required app configuration/changes to work with a backend API server. The examples in the egeria-ui repo all make use of node, but the container we build contains static content only.

I have configured nginx to redirect /api to the correct backend server - this can be verified via:
```
curl -k -i -X GET https://localhost:8443/api/public/app/info
Handling connection for 8443
HTTP/1.1 200
Server: nginx/1.19.0
Date: Tue, 15 Nov 2022 14:20:45 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Vary: origin,access-control-request-method,access-control-request-headers,accept-encoding

{"title":"Egeria Open Metadata | Find the right data with governance","description":"Have a question? || Get in touch via our Slack community https://slack.lfai.foundation/ @@What is Open Metadata? || Find out more on our website https://odpi.github.io/egeria-docs @@Have more cool ideas? || Feel free to let us know your ideas so we can make it better."}%
```

However, whilst the UI will come up with a login screen, the subsequent screen is greyed out.

It looks as if API calls are being made from the browser, and satisfied - ie /api/types is returning 200, however I'm unclear if the content is correct

**Comment by planetf1** [comment]
Current state of testing:

Screenshot 2022-11-15 at 14 43 55

**Comment by planetf1** [comment]
```
10.242.0.27 - - [15/Nov/2022:14:43:16 +0000] "GET /api/public/app/info HTTP/1.1" 200 367 "-" "curl/7.85.0"
10.242.0.27 - - [15/Nov/2022:14:43:36 +0000] "GET / HTTP/1.1" 200 640 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
10.242.0.27 - - [15/Nov/2022:14:43:36 +0000] "GET /manifest.json HTTP/1.1" 200 519 "https://141.125.107.39:8443/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
172.17.27.192 - - [15/Nov/2022:14:43:36 +0000] "GET /static/css/main.639200f8.css HTTP/1.1" 200 191492 "https://141.125.107.39:8443/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
172.17.10.192 - - [15/Nov/2022:14:43:36 +0000] "GET /static/js/main.d0390842.js HTTP/1.1" 200 3993332 "https://141.125.107.39:8443/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
10.242.0.27 - - [15/Nov/2022:14:43:36 +0000] "GET /favicon.ico HTTP/1.1" 200 15406 "https://141.125.107.39:8443/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
172.17.10.192 - - [15/Nov/2022:14:43:36 +0000] "GET /egeria-logo.svg HTTP/1.1" 200 6658 "https://141.125.107.39:8443/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
172.17.27.192 - - [15/Nov/2022:14:43:36 +0000] "GET /api/assets/types HTTP/1.1" 401 0 "https://141.125.107.39:8443/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
10.242.0.27 - - [15/Nov/2022:14:43:37 +0000] "GET /login HTTP/1.1" 200 640 "https://141.125.107.39:8443/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
10.242.0.27 - - [15/Nov/2022:14:43:42 +0000] "POST /api/auth/login HTTP/1.1" 200 0 "https://141.125.107.39:8443/login" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
10.242.0.27 - - [15/Nov/2022:14:43:42 +0000] "GET /api/assets/types HTTP/1.1" 200 0 "https://141.125.107.39:8443/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15"
172.17.27.192 - - [15/Nov/2022:14:43:45 +0000] "GET /Main_Login.asp HTTP/1.1" 200 640 "https://158.175.190.102:8443/" "Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; WOW64; Trident/6.0)"
10.242.0.27 - - [15/Nov/2022:14:43:45 +0000] "GET /Main_Login.asp HTTP/1.1" 200 640 "https://158.175.190.102:8443/" "Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; WOW64; Trident/6.0)"
10.242.0.27 - - [15/Nov/2022:14:43:46 +0000] "POST /login.cgi HTTP/1.1" 405 559 "https://158.175.190.102:8443/" "Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; WOW64; Trident/6.0)"
172.17.10.192 - - [15/Nov/2022:14:43:46 +0000] "POST /login.cgi HTTP/1.1" 405 559 "https://158.175.190.102:8443/" "Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; WOW64; Trident/6.0)"
172.17.10.192 - - [15/Nov/2022:14:43:48 +0000] "GET / HTTP/1.1" 200 640 "-" "Mozilla/5.0 (Windows NT 5.1; rv:9.0.1) Gecko/20100101 Firefox/9.0.1"
➜  charts git:(issue480)
```

**Comment by planetf1** [comment]
Nothing interesting (except confirmation of login) on the ui chassis, no errors in the static ui container. UI hangs.

**Comment by planetf1** [comment]
This problem is now understood.
 - UI hangs due to getting null content instead of json in response
 - UI chassis returns 0 length content due to ..
 - backend call to egeria server returning A permission issue
 - due to setup of userids Using 'faithbroker' for example works, whilst 'admin' does not - this last point isn't a bug as such, more a requirement to document which userids should be used in this scenario
 
 I'll raise issues & merge these charts as a prerelease

**Comment by planetf1** [comment]
Actually cannot - had thought there was a new container image, but it's not available yet (egeria-ui)

File: charts/odpi-egeria-lab/templates/egeria-uistatic.yaml [hunk 1]
```diff
@@ -149,4 +176,40 @@ binaryData:
...
...WG13YTJxClVlM05zZmloV3FSWU1GWXhDekFKQmdOVkJBWVRBbFZUTVFzd0NRWURWUVFJREFKRFFURVJNQThHQTFVRUNnd0kKVEVaQlNVUmhkR0V4SnpBbEJnTlZCQU1NSGtWblpYSnBZVkp2YjNSRFpYSjBhV1pwWTJGMFpVRjFkR2h2Y21sMAplWUlDRUFBd0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dJQkFDbCtpWjdQZC8vY2FHQWJIeHk1d2wrVkl1NDk3SkxJCk1KaVBYa0xzWXQ0N1ZDQUtxZWY0TjZZek8zZ0I4bXZ6VzdxNEFHbFdSR1dEbVJYbnY2R2J6RWxQaE8wTjE5eGoKVWtpT2RkR3ZVb1pwTXVMYmFkQUZjU29MZUNjRlFQMzYrZzgwMktuWDNHZ3M5R2U3aFllSUNrd25hbkFFdFRQWAp3NFBITnFnZzVWY3VGcHdVZ0lyeGl4c0RqbTVZcDhEaDdhaEU3Z3dITFRXWTdLdEdjUS9Qcis4MmI2ZkxnSGd3ClRwSmV6bmdEREozQU4yNUJ1eG5DU2lneUpLcEM2T0pFdkZJMlBkVW85TUZURkhrbHZNWXdEQlFtRlpvOU5YVWkKVURpMzk1cklKdXJjVnVqSlBxMUVGTlVaUkRPZkR2UDRWYmhJR2ZhRHUxU3VMeDMveEVCdGtuVDVvSENPcitGbQpjT1B4WFZ3MXY5bEJFRzFXaGFva2ZaWUl0cFJaOEVlU05RNTl5bjVYV1RZaGRIa1Q5bEhZUDc4TG5hSGZZUENCCm1wT0ozZFJSTCtRNUZ3aUZMTnNNTG5CZHI2OEROT24wN29RaDNmWjFrMGhCQ2VkNi9kYjdydHZVK2NyeU5wVWgKeVJNeFJ6elJ6K0JDSVJxQ2FzNExGaGdTMGNJMDlpVmU3K2xySW5OUUREMURaSjR0MzQ0S21CR0VvNVZQN3lWRgpBT2hnd09lYmdyS2sycTdkbk4rb2t2b0ZIVSt5OEQzMjQ4TXFuby9TT0xRYTROcVJCemhkTjU2UHpxb3lsckEvCmhHaUtWby8ydFlScy9Ub0pvSVRuY0piODBFc1lJc201RWxOMkZoa2tabnlKekVTekRNcXpQK041ZEdmdFp3TFMKcnpMNFBNVEd6YmI5Ci0tLS0tRU5EIFRSVVNURUQgQ0VSVElGSUNBVEUtLS0tLQo=
```
Comment (planetf1) [inline charts/odpi-egeria-lab/templates/egeria-uistatic.yaml:211]: This needs updating to current cert! (as with previous key)

**Comment by planetf1** [comment]
@lpalashevski @sarbull 

We plan to branch earlyish this week and start on v4 testing
If we are to include the new UI in v4 I need a published container image ideally today or tomorrow at the latest
Else we should defer this UI update to a future release

cc: @mandy-chessell @dwolfson

**Review by lpalashevski — APPROVED** [review]

**Pull request merged**